packages/augur-sdk/src/state/getter/Markets.ts

```diff
@@ -414,106 +442,153 @@ export class Markets {
+          extendedSearchOptions.push({
+            field: ["category" + (i + 1)],
+            query: params.categories[i],
+            bool: "and",
+          });
+        }
+        const categoryResults = await db.fullTextMarketSearch(null, extendedSearchOptions);
+        keyedCategoryResults = _.keyBy(
+          categoryResults,
+          (searchResult: MarketFields) =>  { return searchResult.market; }
+        );
+        if (!_.isEmpty(keyedFullTextResults)) {
+          // Reset keyedSearchResults to intersection of searchResults & categoryResults
+          keyedFullTextResults = {};
+          for (let i = 0; i < searchResults.length; i++) {
+            if (categoryResults[searchResults[i].market]) {
+              keyedFullTextResults[searchResults[i].market] = categoryResults[searchResults[i].market];
+            }
+          }
+        } else {
+          keyedFullTextResults = keyedCategoryResults;
+        }
+      }
 
-      filteredKeyedMarketCreatedLogs = Object.entries(
+      filteredKeyedMarketCreatedLogs = Object.values(
         keyedMarketCreatedLogs
       ).reduce((previousValue: any, currentValue: any) => {
-        if (keyedFullTextMarketIds[currentValue[0]]) {
-          previousValue[currentValue[0]] = currentValue[1];
+        if (keyedFullTextResults[currentValue.market]) {
+          previousValue[currentValue.market] = currentValue;
         }
         return previousValue;
       }, []);
     }
 
-    await Promise.all(
-      Object.entries(filteredKeyedMarketCreatedLogs).map(
-        async (marketCreatedLogInfo: any) => {
-          let includeMarket = true;
-
-          if (params.disputeWindow) {
-            const market = await augur.contracts.marketFromAddress(
-              marketCreatedLogInfo[0]
-            );
-            const disputeWindowAddress = await market.getDisputeWindow_();
-            if (params.disputeWindow !== disputeWindowAddress) {
-              includeMarket = false;
-            }
-          }
+    let filteredMarketsInfo: any[] = [];
+    for (const marketCreatedLogInfo of Object.values(filteredKeyedMarketCreatedLogs)) {
+      let includeMarket = true;
 
-          // TODO: when currentOrders event table exists just check that
-          if (params.hasOrders) {
-            const orderCreatedLogs = await db.findOrderCreatedLogs({
-              selector: { market: marketCreatedLogInfo[0] },
-            });
-            const orderCanceledLogs = await db.findOrderCanceledLogs({
-              selector: { market: marketCreatedLogInfo[0] },
-            });
-            const orderFilledLogs = await db.findOrderFilledLogs({
-              selector: { market: marketCreatedLogInfo[0], amount: '0x00' },
-            });
-            if (
-              orderCreatedLogs.length - orderCanceledLogs.length ===
-              orderFilledLogs.length
-            ) {
-              includeMarket = false;
-            }
-          }
+      if (params.disputeWindow) {
```

Comment:
It would be helpful to have disputeWindow information for the MarketInfo if the market has a dispute window.
having startTime and endTime of the dispute window would be awesome. (this can be added in a different PR, created tix <https://github.com/AugurProject/augur/issues/3025>)

Reply:
Sounds good!